pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -102,10 +103,30 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		return cmdutil.UsageError(cmd, "Must pass a name of a BuildConfig or specify build name with '--from-build' flag")
 	}
 
-	name := buildName
-	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	var (
+		name     = buildName
+		resource = ""
+		isBuild  = true
+	)
+
+	if len(name) == 0 && len(args[0]) > 0 {
+		mapper, _ := f.Object()
+		resource, name, err = osutil.ResolveResource("buildconfigs", args[0], mapper)
+		if err != nil {
+			return err
+		}
+		switch resource {
+		case "buildconfigs", "buildconfig":
```

Comment:
if ResolveResource normalizes, which it should, you only have to handle the plural case